src/libsyntax/parse/parser.rs

```diff
@@ -5758,18 +5758,32 @@ impl<'a> Parser<'a> {
                                      vis: Visibility,
                                      attrs: Vec<Attribute> )
                                      -> PResult<'a, StructField> {
+        let mut seen_comma: bool = false;
         let a_var = self.parse_name_and_ty(lo, vis, attrs)?;
+        if self.token == token::Comma {
+            seen_comma = true;
+        }
         match self.token {
             token::Comma => {
                 self.bump();
             }
             token::CloseDelim(token::Brace) => {}
             token::DocComment(_) => {
+                let previous_span = self.prev_span;
                 let mut err = self.span_fatal_err(self.span, Error::UselessDocComment);
                 self.bump(); // consume the doc comment
-                if self.eat(&token::Comma) || self.token == token::CloseDelim(token::Brace) {
+                let comma_after_doc_seen = self.eat(&token::Comma);
+                // `seen_comma` is always false, because we are inside doc block
+                // condition is here to make code more readable
+                if seen_comma == false && comma_after_doc_seen == true{
+                    seen_comma = true;
+                }
+                if comma_after_doc_seen || self.token == token::CloseDelim(token::Brace) {
                     err.emit();
                 } else {
+                    if seen_comma== false {
+                        err.span_suggestion(previous_span, "missing comma here: `,`", "".into());
```

Comment:
this should be
```
err.span_suggestion(sp, "missing comma here", ",".into());
```

The output will be the same, but now third party tools, like VSCode can apply the suggestion.

`sp` should be the result of calling [`codemap.next_point(previous_span)`](https://github.com/rust-lang/rust/blob/75af15ee6ca0c12b699a17984b033363cd25e9c3/src/libsyntax/codemap.rs#L720), otherwise it is pointing at the type, instead of pointing _after_ the type, where the comma is supposed to be.